packages/workspace/src/browser/workspace-service.ts

```diff
@@ -379,18 +387,19 @@ export class WorkspaceService implements FrontendApplicationContribution {
         }
     }
 
-    async spliceRoots(start: number, deleteCount?: number, ...rootsToAdd: URI[]): Promise<URI[]> {
+    async spliceRoots(start: number, deleteCount?: number, ...rootsToAdd: { uri: string, name?: string }[]): Promise<URI[]> {
```

Comment:
It looks nice, but we have to be backward compatible, the new signature should look like this:
```ts
async spliceRoots(start: number, deleteCount?: number, ...rootsToAdd: (URI | { uri: URI, name?: string })[]): Promise<URI[]> {
```
When a raw URI is passed in, it should be treated as a new workspace root without any custom `name`.